xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/pvr/MenuHook.h

```diff
@@ -0,0 +1,110 @@
+/// was set to related type:
+/// - @ref kodi::addon::CInstancePVRClient::CallSettingsMenuHook()
+/// - @ref kodi::addon::CInstancePVRClient::CallChannelMenuHook()
+/// - @ref kodi::addon::CInstancePVRClient::CallEPGMenuHook()
+/// - @ref kodi::addon::CInstancePVRClient::CallRecordingMenuHook()
+/// - @ref kodi::addon::CInstancePVRClient::CallTimerMenuHook()
+///
+//@{
+class PVRMenuhook : public CStructHdl<PVRMenuhook, PVR_MENUHOOK>
+{
+public:
+  /*! \cond PRIVATE */
+  PVRMenuhook()
+  {
+    m_cStructure->iHookId = 0;
+    m_cStructure->iLocalizedStringId = 0;
+    m_cStructure->category = PVR_MENUHOOK_UNKNOWN;
+  }
+  PVRMenuhook(const PVRMenuhook& data) : CStructHdl(data) {}
+  PVRMenuhook(const PVR_MENUHOOK* data) : CStructHdl(data) {}
+  PVRMenuhook(PVR_MENUHOOK* data) : CStructHdl(data) {}
+  /*! \endcond */
+
+  /// @defgroup cpp_kodi_addon_pvr_Defs_Menuhook_PVRMenuhook_Help *Value Help*
+  /// @ingroup cpp_kodi_addon_pvr_Defs_Menuhook_PVRMenuhook
+  /// ----------------------------------------------------------------------------
+  ///
+  /// <b>The following table contains values that can be set with @ref cpp_kodi_addon_pvr_Defs_Menuhook_PVRMenuhook :</b>
+  /// | Name | Type | Set call | Get call | Usage
+  /// |------|------|----------|----------|-----------
+  /// | **This hook's identifier** | `unsigned int` | @ref PVRMenuhook::SetHookId "SetHookId" | @ref PVRMenuhook::GetHookId "GetHookId" | *required to set*
+  /// | **Localized string Identifier** | `unsigned int` | @ref PVRMenuhook::SetLocalizedStringId "SetLocalizedStringId" | @ref PVRMenuhook::GetLocalizedStringId "GetLocalizedStringId" | *required to set*
+  /// | **Category of menu hook** | @ref PVR_MENUHOOK_CAT | @ref PVRMenuhook::SetCategory "SetCategory" | @ref PVRMenuhook::GetCategory "GetCategory" | *required to set*
+
+  /// @addtogroup cpp_kodi_addon_pvr_Defs_Menuhook_PVRMenuhook
+  /// @copydetails cpp_kodi_addon_pvr_Defs_Menuhook_PVRMenuhook_Help
+  //@{
+
+  /// @brief **required**\n
+  /// This hook's identifier.
+  void SetHookId(unsigned int hookId) { m_cStructure->iHookId = hookId; }
+
+  /// @brief To get with @ref SetHookId() changed values.
+  unsigned int GetHookId() const { return m_cStructure->iHookId; }
+
+  /// @brief **required**\n
+  /// The id of the label for this hook in @ref kodi::GetLocalizedString().
+  void SetLocalizedStringId(unsigned int localizedStringId)
+  {
+    m_cStructure->iLocalizedStringId = localizedStringId;
+  }
+
+  /// @brief To get with @ref SetLocalizedStringId() changed values.
+  unsigned int GetLocalizedStringId() const { return m_cStructure->iLocalizedStringId; }
+
+  /// @brief **required**\n
+  /// Category of menu hook.
+  void SetCategory(PVR_MENUHOOK_CAT category) { m_cStructure->category = category; }
+
+  /// @brief To get with @ref SetCategory() changed values.
+  PVR_MENUHOOK_CAT GetCategory() const { return m_cStructure->category; }
```

Comment:
Here, you break encapsulation as you expose a C API data type. This looks wrong.